viite-UI/src/view/RoadAddressBrowserWindow.js

```diff
@@ -233,32 +299,88 @@
                     params.minRoadPartNumber = minRoadPartNumber.value;
                 if (maxRoadPartNumber.value)
                     params.maxRoadPartNumber = maxRoadPartNumber.value;
+                return params;
+            }
 
-                roadAddressCollection.fetchByTargetValue(params);
+            //reset ely and roadAddrStartDate input field's custom validity
+            ely.setCustomValidity("");
+            roadAddrStartDate.setCustomValidity("");
+
+
+            switch (targetValue) {
+                case "Roads":
+                    if (willPassValidations())
+                        roadAddressCollection.fetchByTargetValue(createParams());
+                    else
+                        event.preventDefault();
+                    break;
+                case "Nodes":
+                    if (willPassValidations())
+                        roadAddressCollection.fetchByTargetValue(createParams());
+                    else
+                        event.preventDefault();
+                    break;
+                case "Junctions":
+                    if (willPassValidations())
+                        roadAddressCollection.fetchByTargetValue(createParams());
+                    else
+                        event.preventDefault();
+                    break;
+                case "RoadNames":
+                    validateDate(roadAddrStartDateObject);
+                    if (reportValidations())
+                        roadAddressCollection.fetchByTargetValue(createParams());
+                    else
+                        event.preventDefault();
+                    break;
+                default:
             }
         }
 
-        eventbus.on('roadAddressBrowser:roadsFetched', function () {
+        eventbus.on('roadAddressBrowser:roadsFetched', function (params) {
+            me.setSearchParams(params);
             applicationModel.removeSpinner();
             showResultsForRoads();
         });
 
-        eventbus.on('roadAddressBrowser:nodesFetched', function () {
-           applicationModel.removeSpinner();
-           showResultsForNodes();
+        eventbus.on('roadAddressBrowser:nodesFetched', function (params) {
```

Comment:
(Ja tänne samanlainen "näyttää toisteiselta" -nillinniikommentti. 
... eli jos refaktorointisormia syyhyää, voi ihmetellä, saako nää eventbus.on:it nätisti yhdistettyä.)